src/views/connection.vue

```diff
@@ -1,23 +1,26 @@
 <script>
-import { mapActions, mapState } from 'vuex'
+import { mapActions, mapState } from "vuex";
 
 export default {
-  name: 'Connection',
+  name: "Connection",
   data() {
     return {
-      queryJson: '{}'
-    }
+      queryJson: "{}",
+      showTotalNumberOfResults: false
+    };
   },
   computed: {
-    ...mapState(['collectionName', 'records'])
+    ...mapState(["collectionName", "records"])
   },
   methods: {
-    ...mapActions(['find']),
+    ...mapActions(["find"]),
     doQuery() {
-      this.find(JSON.parse(this.queryJson))
+      this.showTotalNumberOfResults = false;
+      this.find(JSON.parse(this.queryJson));
```

Comment:
This should be an await operation for a meaningful split between `showTotalNumberOfResults = false` and `showTotalNumberOfResults = true`, otherwise the number isn't hidden while querying.

It would be great to implement this in the store — right now we are calling `toArray()` and realizing all the results, however this wouldn't work for large datasets. Instead, we should work with a cursor. MongoDB cursors have a `count` method to return the number of total results for that query, so we should be utilising that.

Reply:
@emredalkiran have you seen this one?

Reply:
@dashersw Yes, just sent a commit. I'm fairly new to Vue.js and MongoDB and I don't know if I did it the correct way. Currently, state.records are still assigned through `toArray` method of `cursor`. I've checked MongoDB documentation for cursor and I think, it will also need to evolve through reading as batches or via streaming that will be triggered with a user interaction like scrolling or pagination.